feat: add support for NavigatorObservers (#35)
JaspervanRiet/duck_router: duck_router/README.md

```dart
DuckRouter.of(context).navigate(to: DialogPageLocation);
```

## Navigator observers

> Adding a solid support for [NavigatorObservers](https://api.flutter.dev/flutter/widgets/NavigatorObserver-class.html) is not trivial, due to the many limitations it has, and the fact we want to hide the implementation details of this package as much as possible.

Reviewer: ```suggestion
Adding support for [NavigatorObservers](https://api.flutter.dev/flutter/widgets/NavigatorObserver-class.html) to the standard we would like is not trivial, due to the many limitations it has, and the fact we want to hide the implementation details of this package as much as possible.
```
